src/NuGet.Core/NuGet.Packaging/Signing/Signatures/Signature.cs

```diff
@@ -17,35 +17,49 @@ namespace NuGet.Packaging.Signing
     /// </summary>
     public class Signature
     {
-        private readonly byte[] _data;
 
-        private Signature(byte[] data)
-        {
-            _data = data ?? throw new ArgumentNullException(nameof(data));
-        }
+#if IS_DESKTOP
 
         /// <summary>
-        /// Indicates if this is an author or repository signature.
+        /// A SignedCms object holding the signature and SignerInfo.
         /// </summary>
-        public SignatureType Type { get; private set; }
+        public SignedCms SignedCms { get; }
 
         /// <summary>
-        /// Additional counter signatures.
+        /// Index to the SignerInfo object in the SignedCms.SignInfos.
+        /// The index is used to indicate the SignerInfo for this Signature.
         /// </summary>
-        public IReadOnlyList<Signature> AdditionalSignatures { get; private set; } = (new List<Signature>()).AsReadOnly();
+        public int SignerInfoIndex { get; }
```

Comment:
Is this ever non-zero? We should avoid exposing it if possible.

//cc @PatoBeltran

Reply:
Not currently, but this can be non zero as we add support for multiple signatures

Reply:
As the current spec describes, this value will always be zero. If the spec changes to allow multiple signatures (I seriously doubt it because of the complexity it would bring to the verify code) then it could be other than zero.

Reply:
🔪 kill it